src/Compilers/CSharp/Portable/Symbols/Source/SourceEventSymbol.cs

```diff
@@ -436,13 +436,22 @@ private void CheckAccessibility(Location location, BindingDiagnosticBag diagnost
                                                    bool isFieldLike, Location location,
                                                    BindingDiagnosticBag diagnostics, out bool modifierErrors)
         {
-            bool isInterface = this.ContainingType.IsInterface;
-            var defaultAccess = isInterface && !explicitInterfaceImplementation ? DeclarationModifiers.Public : DeclarationModifiers.Private;
+            bool inInterface = this.ContainingType.IsInterface;
+            bool inExtension = this.ContainingType.IsExtension;
+            var defaultAccess = inInterface && !explicitInterfaceImplementation ? DeclarationModifiers.Public : DeclarationModifiers.Private;
             var defaultInterfaceImplementationModifiers = DeclarationModifiers.None;
 
             // Check that the set of modifiers is allowed
             var allowedModifiers = DeclarationModifiers.Unsafe;
-            if (!explicitInterfaceImplementation)
+            if (inExtension)
+            {
+                allowedModifiers |= DeclarationModifiers.New |
+                                    DeclarationModifiers.Static |
+                                    DeclarationModifiers.Private |
+                                    DeclarationModifiers.Internal |
+                                    DeclarationModifiers.Public;
```

Comment:
>DeclarationModifiers.Public [](http://example.com/codeflow?start=36&length=27)

Why are we excluding protected flavors?

Reply:
Hmm, I thought we'd disallowed all protected flavors, but going back to the spec and notes, that's not the case... Thanks